dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResourceTest.java

```diff
@@ -0,0 +1,428 @@
+        final ServiceIntegrationSiteView serviceIntegrationHostView2 = (ServiceIntegrationSiteView) responseEntityView2.getEntity();
+        final ServiceIntegrationView serviceIntegrationView2 = serviceIntegrationHostView2.getService();
+        Assert.assertEquals(1, serviceIntegrationView2.getConfigurationsCount());
+        Assert.assertEquals("lola", serviceIntegrationView2.getName());
+        final List<SiteView> hosts = serviceIntegrationHostView2.getSites();
+        Assert.assertNotNull(hosts);
+        Assert.assertFalse(hosts.isEmpty());
+        Assert.assertEquals(hosts.get(0).getSiteId(),host.getIdentifier());
+        Assert.assertTrue(
+                hosts.stream().anyMatch(hostView -> host.getIdentifier().equals(hostView.getSiteId()))
+                );
+
+        final Response detailedIntegrationResponse = serviceIntegrationResource.getDetailedServiceIntegration(request, response, serviceKey, host.getIdentifier());
+        Assert.assertEquals(HttpStatus.SC_OK, detailedIntegrationResponse.getStatus());
+        final ResponseEntityView responseEntityView3 = (ResponseEntityView) detailedIntegrationResponse.getEntity();
+        final ServiceIntegrationDetailedView serviceIntegrationDetailedView = (ServiceIntegrationDetailedView) responseEntityView3.getEntity();
+        Assert.assertEquals(serviceIntegrationDetailedView.getHost().getSiteId(),host.getIdentifier());
+
+        final Response deleteIntegrationsResponse = serviceIntegrationResource.deleteAllServiceIntegrationSecrets(request, response, serviceKey, host.getIdentifier());
+        Assert.assertEquals(HttpStatus.SC_OK, deleteIntegrationsResponse.getStatus());
+
+        final Response detailedIntegrationResponseAfterDelete = serviceIntegrationResource.getDetailedServiceIntegration(request, response, serviceKey, host.getIdentifier());
+        Assert.assertEquals(HttpStatus.SC_NOT_FOUND, detailedIntegrationResponseAfterDelete.getStatus());
+
+    }
+
+    @Test
+    public void  Test_Create_service_descriptor_Then_Create_Service_Integration_Then_Delete_One_Single_Secret() throws IOException {
+
+        final Map<String, Param> paramMap = ImmutableMap.of(
+                "param1", Param.newParam("val-1", false, Type.STRING, "label", "hint", true),
+                "param2", Param.newParam("val-2", false, Type.STRING, "label", "hint", true),
+                "param3", Param.newParam("val-3", false, Type.STRING, "label", "hint", true)
+        );
+        final Host host = new SiteDataGen().nextPersisted();
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        final HttpServletResponse response = mock(HttpServletResponse.class);
+        final String serviceKey = String.format("lol_%d",System.currentTimeMillis());
+        final String fileName = String.format("%s.yml",serviceKey);
+        final InputStream inputStream = createServiceDescriptorFile(fileName, serviceKey, "lola",
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ensure that resources like this InputStream object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)